### PR TITLE
New exercice proposal to use maps in the piscine

### PR DESCRIPTION
**Pull request by Clemalfroy**
Tested and written with @Frenchris

### CODE REVIEW

File: solutions/maxwordcountn.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package solutions
+
+import (
+	"sort"
+	"strings"
+)
+
+type KeyValue struct {
```
Comment (xpetit) [inline solutions/maxwordcountn.go:8]: `KeyValue` and its members don't need to be exported

File: tests/maxwordcountn_test/main.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	"fmt"
+	"strings"
+	student "student"
+
+	"github.com/01-edu/go-tests/lib/challenge"
+	"github.com/01-edu/go-tests/lib/random"
+	"github.com/01-edu/go-tests/solutions"
+)
+
+type Args struct {
```
Comment (xpetit) [inline tests/maxwordcountn_test/main.go:13]: Doesn't need to be exported

**Review by xpetit — CHANGES_REQUESTED** [review]
Good code, just reformat those files with `gofumpt` please

File: solutions/maxwordcountn.go (hunk 1 above)
Comment (Clemalfroy) [reply solutions/maxwordcountn.go]: replaced by `keyValue`

File: tests/maxwordcountn_test/main.go (hunk 2 above)
Comment (Clemalfroy) [reply tests/maxwordcountn_test/main.go]: replaced by `functionArgs`

**Review by xpetit — APPROVED** [review]

**Pull request merged**